packages/application-config/src/load-config.ts

```diff
@@ -1,83 +1,86 @@
-import { execFileSync } from 'node:child_process';
 import fs from 'node:fs';
 import path from 'node:path';
-import { cosmiconfigSync, defaultLoaders, type LoaderSync } from 'cosmiconfig';
+import {
+  cosmiconfig,
+  defaultLoaders,
+  defaultLoadersSync,
+  Loader,
+} from 'cosmiconfig';
+import { TypeScriptLoader } from 'cosmiconfig-typescript-loader';
 import { MissingOrInvalidConfigError } from './errors';
 import type { JSONSchemaForCustomApplicationConfigurationFiles } from './schemas/generated/custom-application.schema';
 import type { JSONSchemaForCustomViewConfigurationFiles } from './schemas/generated/custom-view.schema';
 
-function doesFileExist(path: string): boolean {
+// See the following issues for more context, contributing to failing Jest tests:
+//  - Issue: https://github.com/nodejs/node/issues/40058
+//  - Resolution: https://github.com/nodejs/node/pull/48510 (Node v20.8.0)
+// Copied from @commitlint/load/src/utils/load-config.ts
+export const isDynamicAwaitSupported = () => {
+  const [major, minor] = process.version
+    .replace('v', '')
+    .split('.')
+    .map((val) => parseInt(val));
+
+  return major >= 20 && minor >= 8;
+};
+
+// Is the given directory set up to use ESM (ECMAScript Modules)?
+// Copied from @commitlint/load/src/utils/load-config.ts
+export const isEsmModule = async (cwd: string) => {
+  const packagePath = path.join(cwd, 'package.json');
+
   try {
-    fs.accessSync(path);
-    return true;
+    await fs.promises.access(packagePath, fs.constants.F_OK);
```

Comment:
**nit**: do we actually need this `access` call?
It seems it will throw and error if user does not have access but the next line (read file) would also throw an error when no access is available and we're catching both in the same `catch` statement.

Reply:
True, we can omit the `access` call yes

Reply:
https://github.com/commercetools/merchant-center-application-kit/pull/3669/commits/fe0e58007bc6394dd54bee3e9de38674faad1cf9